hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieCompactionConfig.java

```diff
@@ -146,21 +146,12 @@ public class HoodieCompactionConfig extends HoodieConfig {
           + "compaction during each compaction run. By default. Hudi picks the log file "
           + "with most accumulated unmerged data");
 
-  public static final ConfigProperty<String> COMPACTION_LAZY_BLOCK_READ_ENABLE = ConfigProperty
-      .key("hoodie.compaction.lazy.block.read")
-      .defaultValue("true")
-      .markAdvanced()
-      .withDocumentation("When merging the delta log files, this config helps to choose whether the log blocks "
-          + "should be read lazily or not. Choose true to use lazy block reading (low memory usage, but incurs seeks to each block"
-          + " header) or false for immediate block read (higher memory usage)");
-
-  public static final ConfigProperty<String> COMPACTION_REVERSE_LOG_READ_ENABLE = ConfigProperty
-      .key("hoodie.compaction.reverse.log.read")
-      .defaultValue("false")
-      .markAdvanced()
-      .withDocumentation("HoodieLogFormatReader reads a logfile in the forward direction starting from pos=0 to pos=file_length. "
-          + "If this config is set to true, the reader reads the logfile in reverse direction, from pos=file_length to pos=0");
+  // These config properties are kept to make sure defaults are properly propagated
+  public static final ConfigProperty<String> COMPACTION_LAZY_BLOCK_READ_ENABLE =
```

Comment:
What does it mean for "properly propagated", I didn't see any references to these variables from the code.

Reply:
The propagation of default values are done using Java reflections.  See `HoodieCompactionConfig$Builder#build`
```
    public HoodieCompactionConfig build() {
      compactionConfig.setDefaults(HoodieCompactionConfig.class.getName());
      return compactionConfig;
    }
```
which checks the declared fields as below:
```
  protected void setDefaults(String configClassName) {
    Class<?> configClass = ReflectionUtils.getClass(configClassName);
    Arrays.stream(configClass.getDeclaredFields())
        .filter(f -> Modifier.isStatic(f.getModifiers()))
        .filter(f -> f.getType().isAssignableFrom(ConfigProperty.class))
        .forEach(f -> {
          try {
            ConfigProperty<?> cfgProp = (ConfigProperty<?>) f.get("null");
            if (cfgProp.hasDefaultValue() || cfgProp.hasInferFunction()) {
              setDefaultValue(cfgProp);
            }
          } catch (IllegalAccessException e) {
            e.printStackTrace();
          }
        });
  }
```

Reply:
Can we do `compactionConfig.setDefaults(HoodieMemoryConfig.class.getName());` additionally ?

Reply:
I think you mean `compactionConfig.setDefaults(HoodieReaderConfig.class.getName());`. Makes sense.  It's fixed now.